core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -503,8 +503,8 @@ private void updateCacheStream(long blockId) {
     }
     Preconditions.checkNotNull(mCurrentBlockInStream, "mCurrentBlockInStream");
 
-    // do not create cache stream when the block is in local worker
-    if (!mShouldCache || mCurrentBlockInStream.Source() == BlockInStreamSource.LOCAL) {
+    // do not create cache stream when the block is not in remote worker
```

Comment:
please javadoc here why we don't create cache stream if the block is read from UFS